multimodal/src/autogluon/multimodal/utils/object_detection.py

```diff
@@ -658,12 +725,379 @@ def get_coco_format_classes(sample_data_path):
+        root = download(root)
+    rpath = Path(root).expanduser()
+
+    labels_file = os.path.join(rpath, "labels.txt")
+    if os.path.exists(labels_file):
+        with open(labels_file) as f:
+            class_names = [line.rstrip().lower() for line in f]
+        print(f"using class_names in labels.txt: {class_names}")
+    else:
+        ## read the class names and save results
+        logger.warning(
+            "labels.txt does not exist, using default VOC names. "
+            "To create labels.txt, run ls Annotations/* > pathlist.txt in root dir"
+        )
+        class_names = dump_voc_classes(
+            voc_annotation_path=os.path.join(root, "Annotations"), voc_class_names_output_path=labels_file
+        )
+
+    return class_names
 
 
 def get_detection_classes(sample_data_path):
     if os.path.isdir(sample_data_path):
         return get_voc_format_classes(sample_data_path)
     else:
         return get_coco_format_classes(sample_data_path)
+
+
+def visualize_detection(
+    pred: Iterable,
+    data: Union[pd.DataFrame, Dict],
+    detection_classes: List[str],
+    conf_threshold: float,
+    visualization_result_dir: str,
+) -> List[np.ndarray]:
+    """
+    Visualize detection results for one image, and save to visualization_result_dir
+
+    Parameters
+    ----------
+    pred
+        List containing detection results
+    data
+        The image info for the testing images
+    detection_classes
+        All classes for detection
+    conf_threshold
+        Bounding box confidence threshold to filter unwanted detections
+    visualization_result_dir
+        Directory to save the visualization results
+    Returns
+    -------
+    an List of np.ndarray of visualized images
+    """
+    try:
+        import cv2
+    except:
+        raise ImportError("No module named: cv2. Please install cv2 by 'pip install cv2'")
+
+    if isinstance(data, dict):
+        image_paths = data["image"]
+    else:
+        image_paths = data["image"].to_list()
+    idx2classname = {i: classname for (i, classname) in enumerate(detection_classes)}
+    visualized_images = []
+    for image_pred, image_path in zip(pred, image_paths):
+        im = cv2.imread(image_path)
+        tlwhs = []
+        obj_ids = []
+        conf_scores = []
+        for idx, per_cls_bboxes in enumerate(image_pred):
+            for bbox in per_cls_bboxes:
+                ## x1, y1, x2, y2, conf_score
+                if bbox[4] > conf_threshold:
+                    tlwhs.append(bbox_xyxy_to_xywh(list(bbox[:4])))
+                    obj_ids.append(idx)
+                    conf_scores.append(bbox[4])
+
+        visualized_im = plot_detections(im, tlwhs, obj_ids, idx2classname, conf_threshold, scores=conf_scores)
+        visualized_images.append(visualized_im)
+        imgname = os.path.basename(image_path)
+        cv2.imwrite(os.path.join(visualization_result_dir, imgname), visualized_im)
+    return visualized_images
+
+
+def plot_detections(
```

Comment:
I think mmdetection has visualization support, https://mmdetection.readthedocs.io/en/latest/_modules/mmdet/apis/inference.html#show_result_pyplot. What is the difference between our version with theirs? Can we simply adopt their implementation?

Reply:
If we use mmdet visualization, the input will be predictor._model.model, and it's a bit confusing to user. Maybe we can add a wrapper for it. I think for now we can keep current visualization, marking a TODO here, and after the release we shall check if we would like to use their implementation.

Reply:
Sure, let's do this after the release, thank you.

Reply:
@FANGAreNotGnu @yongxinw Shall we use `mmcv.imshow_bboxes(data, result, top_k=top_k, **kwargs)` ? https://github.com/open-mmlab/mmcv/blob/6254ebef8d1d106e9cca8d4097786c6172cf70ce/mmcv/visualization/image.py#L37

Reply:
Per offline discussion, we can add a TODO.

Reply:
Sure will add it as a TODO